include/zephyr/net/http/table.h

```diff
@@ -0,0 +1,107 @@
+/** @file
+ * @brief HTTP table
+ */
+
+/*
+ * Copyright (c) 2023 Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_TABLE_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_TABLE_H_
+
+#include <stdint.h>
+
+#include <zephyr/net/http/server.h>
+
+/**
+ * @brief HTTP table
+ * @defgroup http_table HTTP table
+ * @ingroup networking
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum http_table_key {
```

Comment:
Might be worth it to change `HTTP_TABLE` to `HTTP_HPACK_KEY` with
```shell
sed -i -e 's/HTTP_TABLE/HTTP_HPACK/g' <filename>
```

```suggestion
enum http_hpack_static_key {
```